presto-main/src/main/java/com/facebook/presto/server/AsyncPageTransportServlet.java

```diff
@@ -0,0 +1,193 @@
+import static com.facebook.presto.client.PrestoHeaders.PRESTO_PAGE_TOKEN;
+import static com.facebook.presto.client.PrestoHeaders.PRESTO_TASK_INSTANCE_ID;
+import static com.facebook.presto.server.SerializedPageWriteListener.PAGE_METADATA_SIZE;
+import static com.facebook.presto.util.TaskUtils.DEFAULT_MAX_WAIT_TIME;
+import static com.facebook.presto.util.TaskUtils.randomizeWaitTime;
+import static com.google.common.net.HttpHeaders.CONTENT_LENGTH;
+import static com.google.common.net.HttpHeaders.CONTENT_TYPE;
+import static com.google.common.util.concurrent.Futures.addCallback;
+import static java.lang.Long.parseLong;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static javax.servlet.http.HttpServletResponse.SC_BAD_REQUEST;
+import static javax.servlet.http.HttpServletResponse.SC_INTERNAL_SERVER_ERROR;
+import static javax.servlet.http.HttpServletResponse.SC_NO_CONTENT;
+
+public class AsyncPageTransportServlet
+        extends HttpServlet
+{
+    private static final Logger log = Logger.get(AsyncPageTransportServlet.class);
+
+    private static final Duration PAGE_RESPONSE_WAIT_TIME = new Duration(60, SECONDS);
+
+    private final TaskManager taskManager;
+    private final Executor responseExecutor;
+    private final ScheduledExecutorService timeoutExecutor;
+
+    @Inject
+    public AsyncPageTransportServlet(
+            TaskManager taskManager,
+            @ForAsyncRpc BoundedExecutor responseExecutor,
+            @ForAsyncRpc ScheduledExecutorService timeoutExecutor)
+    {
+        this.taskManager = requireNonNull(taskManager, "taskManager is null");
+        this.responseExecutor = requireNonNull(responseExecutor, "responseExecutor is null");
+        this.timeoutExecutor = requireNonNull(timeoutExecutor, "timeoutExecutor is null");
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response)
+            throws IOException
+    {
+        String requestURI = request.getRequestURI();
+        // Example:  /v1/task/async/{taskId}/results/{bufferId}/{token}
+        List<String> requestURIParts = Arrays.asList(requestURI.split("/"));
+
+        if (requestURIParts.size() != 8) {
+            response.sendError(SC_BAD_REQUEST, format("Unexpected URI for task result request in async mode: %s", requestURI));
+            return;
+        }
+
+        TaskId taskId = TaskId.valueOf(requestURIParts.get(4));
+        OutputBufferId bufferId = OutputBufferId.fromString(requestURIParts.get(6));
+        long token = parseLong(requestURIParts.get(7));
+        DataSize maxSize = DataSize.valueOf(request.getHeader(PRESTO_MAX_SIZE));
+
+        AsyncContext asyncContext = request.startAsync(request, response);
+
+        Duration waitTime = randomizeWaitTime(DEFAULT_MAX_WAIT_TIME);
+        asyncContext.setTimeout(waitTime.toMillis() + PAGE_RESPONSE_WAIT_TIME.toMillis());
```

Comment:
What's this `PAGE_RESPONSE_WAIT_TIME`? (Is this specific to `AyncPageTransport`?

Reply:
> What's this `PAGE_RESPONSE_WAIT_TIME`? (Is this specific to `AyncPageTransport`?

Yes, I have changed this constant timeout to a configurable variable `pageTransportTimeout`